File: app/src/main/java/com/young/minor/livetool/utils/FieldUtil.java
package com.young.minor.livetool.utils;

import java.lang.reflect.Field;

/**
 * created by yf on 2019/3/7.
 */
public class FieldUtil {
    public static Object getField(Class clazz, Object target, String name) throws Exception {
        Field field = clazz.getDeclaredField(name);
        field.setAccessible(true);
        return field.get(target);
    }
    public static Field getField(Class clazz, String name) throws Exception{
        Field field = clazz.getDeclaredField(name);
        field.setAccessible(true);
        return field;
    }
    public static void setField(Class clazz, Object target, String name, Object value) throws Exception {
        Field field = clazz.getDeclaredField(name);
        field.setAccessible(true);
        field.set(target, value);
    }
}

File: app/src/main/java/com/cgtn/minor/liveminority/utils/FieldUtil.kt
package com.cgtn.minor.liveminority.utils

import java.lang.reflect.Field
import java.util.*

/**
 * 目的是为了对字段进行反射操作
 */
class FieldUtil{
    companion object {
        fun <T>getField(clazz: Class<T>, target: Any?, name:String): Any? {
             val filed = clazz.getDeclaredField(name)
             filed.isAccessible = true
             return filed.get(target)
         }

        fun <T> getField(clazz: Class<T>,name: String): Field {
            val filed = clazz.getDeclaredField(name)
            filed.isAccessible = true
            return filed
        }

        fun <T>setField(clazz: Class<T>,target: Any,name: String,value: Any){
            val filed = clazz.getDeclaredField(name)
            filed.isAccessible =true
            filed.set(target,value      )
        }
    }
}